### ISSUE TITLE
Really weird 'ReferenceError: window is not defined'

### ISSUE THREAD

**Issue by Hrishikesh-K** [gatsbyjs/gatsby#25238]
## Description

I have a website up and running already. I had used a starter, but, at this stage, it's more like a completely new website due to a lot of customizations. For the past few days, I didn't have to update my website. But, today, when I got to update it (just added a custom plugin, no other change in source code), I thought it's a nice time to update all dependencies to the latest version. So, I run the update and my website has reached into hell since then. First, it was a problem with a wrongly published version of a plugin, which a few had reported on its repo and it got fixed in a few hours. I thought, now I can finally build the website, but, I was badly mistaken. `gatsby build` is just giving me the error `"window" is not available during server side rendering.`. I've read a lot about it now and I know it's not a Gatsby problem, but, I really have no clue about what's causing this. After a lot of other troubleshooting like adding some code to `gatsby-node.js`, etc., I finally gave up and thought, returning to the last backup of the project is probably the best solution. This last backup I had in my external hard drive is from the time I had successfully published my build online. But, now even that, with those same version of dependencies used, gives the same error. I'm so lost and really worried as I'm not so much of a programmer or someone myself. I do have made 2 to 3 websites with Gatsby, but, they were all based on starters by some kind souls.

### Steps to reproduce

I genuinely have no idea. The error has just started popping up out of a sudden.

### Expected result

The build should work just fine.

### Actual result

`gatsby develop` runs completely fine, no errors, nothing.

The complete log after I run `gatsby build`:
```
success open and validate gatsby-configs - 0.275s
success load plugins - 7.455s
success onPreInit - 0.072s
success delete html and css files from previous builds - 0.097s
success initialize cache - 0.130s
success copy gatsby files - 0.527s
success onPreBootstrap - 0.065s
success createSchemaCustomization - 0.217s
success source and transform nodes - 0.771s
success building schema - 2.572s
success createPages - 0.503s
success createPagesStatefully - 0.516s
success updating schema - 0.212s
success onPreExtractQueries - 0.013s
success extract queries from components - 2.102s
success write out redirect data - 0.017s
success Build manifest and related icons - 0.612s
success onPostBootstrap - 0.663s
info bootstrap finished - 47.469s
success run static queries - 0.258s - 9/9 34.86/s
success run page queries - 1.484s - 13/13 8.76/s
success write out requires - 0.040s
success Building production JavaScript and CSS bundles - 96.260s
success Rewriting compilation hashes - 0.043s
failed Building static HTML for pages - 26.230s

 ERROR #95312 

"window" is not available during server side rendering.

See our docs page for more info on this error: https://gatsby.dev/debug-html

  WebpackError: ReferenceError: window is not defined

  - build-html.js:107 doBuildPages
    [HriTech]/[gatsby]/dist/commands/build-html.js:107:24

  - build-html.js:121 async buildHTML
    [HriTech]/[gatsby]/dist/commands/build-html.js:121:3

  - build.js:219 async build
    [HriTech]/[gatsby]/dist/commands/build.js:219:5
```

### Environment

```
  System:
    OS: Windows 10 10.0.19041
    CPU: (8) x64 Intel(R) Core(TM) i5-8300H CPU @ 2.30GHz
  Binaries:
    Node: 12.17.0 - C:\Program Files\nodejs\node.EXE
    npm: 6.14.4 - C:\Program Files\nodejs\npm.CMD
  Languages:
    Python: 3.8.3
  Browsers:
    Edge: 44.19041.1.0
  npmPackages:
    gatsby: ^2.23.10 => 2.23.10
    gatsby-plugin-canonical-urls: ^2.3.5 => 2.3.5
    gatsby-plugin-manifest: ^2.4.13 => 2.4.13
    gatsby-plugin-react-helmet: ^3.3.5 => 3.3.5
    gatsby-plugin-sitemap: ^2.4.6 => 2.4.6
    gatsby-plugin-sw: ^1.0.1 => 1.0.1
    gatsby-plugin-transition-link: ^1.20.0 => 1.20.0
    gatsby-remark-custom-blocks: ^2.3.5 => 2.3.5
    gatsby-remark-external-links: 0.0.4 => 0.0.4
    gatsby-source-filesystem: ^2.3.13 => 2.3.13
    gatsby-transformer-remark: ^2.8.18 => 2.8.18
  npmGlobalPackages:
    gatsby-cli: 2.12.49
```
I don't have a repo (I'm not very familiar with managing Gits), but, here's the complete source of my website: [HriTech.zip](https://github.com/gatsbyjs/gatsby/files/4821642/HriTech.zip)

This website is really super important to me, so, any help is greatly appreciated.

**Comment by Hrishikesh-K** [comment]
Update: Another user of the previously problematic plugin is mentioning the same error here: https://github.com/TylerBarnes/gatsby-plugin-transition-link/issues/217

Maybe, the plugin is still a problem after all. It's weird, because even when I used the older version from my hard drive, it was still giving the same problem, but, still there's hope. I'll re-open this issue if that plugin isn't a problem.

**Comment by TylerBarnes** [comment]
Thanks for opening this issue, checking it out now!

**Comment by TylerBarnes** [comment]
This is fixed in `gatsby-plugin-transition-link@1.20.2`
Please let me know if you run into any more issues and I'll have a look right away.
I'm rusty on this package since it's been a while since I've been working on it. Looks like my first couple releases did more harm than good 😜 but now I'm back in the hang of it
